x-pack/plugins/data_visualizer/public/application/index_data_visualizer/locator/locator.ts

```diff
@@ -121,6 +145,15 @@ export class IndexDataVisualizerLocatorDefinition
     if (visibleFieldNames) appState.visibleFieldNames = visibleFieldNames;
     if (visibleFieldTypes) appState.visibleFieldTypes = visibleFieldTypes;
 
+    if (showAllFields) appState.showAllFields = showAllFields;
+    if (showEmptyFields) appState.showEmptyFields = showEmptyFields;
+    if (pageSize) appState.pageSize = pageSize;
+    if (sortDirection) appState.sortDirection = sortDirection;
+    if (samplerShardSize) appState.samplerShardSize = samplerShardSize;
+    if (pageIndex) appState.pageIndex = pageIndex;
+    if (sortField) appState.sortField = sortField;
+    if (showDistributions) appState.sortField = sortField;
```

Comment:
Looks like a typo here - `appState.showDistributions = showDistributions;`? Although when I tested this locally with a session saved with `"showDistributions": false`, the view opens with the distributions showing. 

If this is difficult to restore, does it make sense to only persist properties in the session config which affect the search itself, rather than just the view?

Reply:
Thanks for catching that. Fixed the issue with show distributions not registering correctly here  https://github.com/elastic/kibana/pull/121053/commits/a2359f1bc1e9630f67965f07fc3b7f667dfc0ba4